composites/Plugin/SnippetEditor/serialization.js

```diff
@@ -64,69 +84,105 @@ export function serializeEditor( rawContent ) {
  * @returns {Object} The serialized entity.
  */
 export function unserializeEntity( key, name, offset ) {
-	const length = name.length;
-
 	const entityRange = {
 		key,
 		offset,
-		length,
+		length: name.length,
 	};
 
 	const mappedEntity = {
 		data: {
-			mention: new Map( [
-				[ "name", name ],
-				[ "description", "%%" + name + "%%" ],
-			] ),
+			mention: {
+				name,
+			},
 		},
 		mutability: "IMMUTABLE",
-		type: "%%mention",
+		type: "%mention",
 	};
 
 	return { entityRange, mappedEntity };
 }
 
+/**
+ * Find all indices of a search term in a string.
+ *
+ * @param {string} searchTerm The term to search for.
+ * @param {string} text       The text to search in.
+ *
+ * @returns {Array} Array of found indices.
+ */
+const getIndicesOf = ( searchTerm, text ) => {
+	if ( searchTerm.length === 0 ) {
+		return [];
+	}
+
+	let startIndex = 0;
+	let index;
+	const indices = [];
+
+	while ( ( index = text.indexOf( searchTerm, startIndex ) ) > -1 ) {
+		indices.push( index );
+		startIndex = index + searchTerm.length;
+	}
+
+	return indices;
+};
+
 /**
  * Unserializes a piece of content into DraftJS data.
  *
  * @param {string} content The content to unserialize.
+ * @param {Array} tags The tags for the DraftJS mention plugin.
+ *
  * @returns {Object} The raw data ready for convertFromRaw.
  */
-export function unserializeEditor( content ) {
+export function unserializeEditor( content, tags ) {
```

Comment:
As discussed in person. In the future, we're gonna want to extract some parts of this function to make ik more comprehensive.